scheduler-server/src/main/java/gov/nasa/jpl/aerie/scheduler/server/remotes/postgres/ClaimRequestAction.java

```diff
@@ -0,0 +1,38 @@
+package gov.nasa.jpl.aerie.scheduler.server.remotes.postgres;
+
+import java.sql.Connection;
+import java.sql.PreparedStatement;
+import java.sql.SQLException;
+import org.intellij.lang.annotations.Language;
+
+/*package local*/ public class ClaimRequestAction implements AutoCloseable {
+  private static final @Language("SQL") String sql = """
+    update scheduling_request
+      set
+        status = 'incomplete'
+      where (specification_id = ? and status = 'pending');
+  """;
+
+  private final PreparedStatement statement;
+
+  public ClaimRequestAction(final Connection connection) throws SQLException {
+    this.statement = connection.prepareStatement(sql);
+  }
+
+  public void apply(final long specificationId) throws SQLException, UnclaimableRequestException {
+    this.statement.setLong(1, specificationId);
+
+    final var count = this.statement.executeUpdate();
+    if (count < 1) {
+      throw new UnclaimableRequestException(specificationId);
```

Comment:
Since this is an intentional race condition, I'd prefer not to include `throw`s on the happy path. Can you change this to return a boolean instead of throw?

Reply:
I completely agree with this change but I'd like to do this as part of a follow-up PR that makes this change in both `merlin-worker` and `scheduler-worker`. I've avoided making refactoring changes to `merlin-worker` in this PR and would like to keep both workers in lock-step w.r.t. changes as well as we can. :)

Reply:
I'll note this in the future work section and will make sure to capture this in a ticket.